src/prisma/generator/models.py

```diff
@@ -586,9 +586,8 @@ def scalar_type_validator(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         kind = values.get('kind')
         type_ = values.get('type')
 
-        if kind == 'scalar':
-            if type_ is not None and type_ not in TYPE_MAPPING:
-                raise ValueError(f'Unsupported scalar field type: {type_}')
+        if kind == 'scalar' and type_ is not None and type_ not in TYPE_MAPPING:
+            raise ValueError(f'Unsupported scalar field type: {type_}')
```

Comment:
I prefer the original style as it makes it easier to add more cases in the future.

For example, if we needed to add another check for scalar types then with the old style we would just have to add a new nested if statement, however with this new style we will either have to duplicate the `kind == 'scalar'` check or extract it to it's own if statement, bringing the code back to its original state.